packages/shared/src/components/onboarding/PWA/useInstallPWA.ts

```diff
@@ -0,0 +1,37 @@
+import { isPWA } from '../../../lib/func';
+
+interface IBeforeInstallPromptEvent extends Event {
+  prompt: () => Promise<'accepted' | 'dismissed'>;
+  userChoice: Promise<{ outcome: 'accepted' | 'dismissed' }>;
+}
+
+export interface UseInstallPWA {
+  isAvailable: boolean;
+  isCurrentPWA: boolean;
+  promptToInstall: (() => Promise<void | 'accepted' | 'dismissed'>) | null;
+}
+
+let installEvent: IBeforeInstallPromptEvent | null = null;
+globalThis?.addEventListener?.(
+  'beforeinstallprompt',
+  (e: IBeforeInstallPromptEvent) => {
+    e.preventDefault();
+    installEvent = e;
+  },
+  { once: true },
+);
```

Comment:
This is very odd but seems that is [the only supported way to do it](https://developer.mozilla.org/en-US/docs/Web/Progressive_web_apps/Guides/Making_PWAs_installable#triggering_the_install_prompt)